mala/targets/calculation_helpers.py

```diff
@@ -62,8 +62,25 @@ def fermi_function(energy, fermi_energy, temperature):
         Value of the Fermi function.
 
     """
-    return 1.0 / (1.0 + np.exp((energy - fermi_energy) /
-                               (kB * temperature)))
+    exponent = (energy - fermi_energy) / (kB * temperature)
+
+    # Maximum exponent that will not result in inf when performing
+    # np.exp() (for double values only).
+    # Very large exponents result in very large values, and if we don't check
+    # for overflow, infinity. Cutting this back to the maximum float value
+    # (~ 1e308) is more then enough, since we divide by the exponent, and
+    # therefore get zero (or machine precision within zero) either way.
+    # Adding this check removes overflow warnings.
+    # The overhead for inference should be minimal.
+    # Since this function works both for arrays and scalars,
+    # we have to check which maximum to use.
+    if type(energy) == np.ndarray:
+        max_exponent = np.log(np.finfo(exponent.dtype.name).max)
+        exponent[np.argwhere(exponent > max_exponent)] = max_exponent
```

Comment:
Hmm, is `exponent` 1d, so shape `(N,)` ? `argwhere adds an extra dimension, not sure of that's what you want.

```py
>>> a=np.random.rand(100)
>>> a[np.argwhere(a>0.5)].shape
(51, 1)
>>> a[a>0.5].shape
(51,)
```

Why not use the bood mask `exponent > max_exponent` directly?

Reply:
Ah yeah, that is much better! Implemented in 3f7bf58fa9f2a7544df45eecb158ce63f954dee1.